test/python/sdpa/frost/test_sdpa_fwd_split_kv_sm100.py

```
@@ -1287,3 +1293,152 @@ def test_api_fp8_amax_describes_the_recombined_output(mx, d):
+def test_quantized_split_reduces_in_half_not_in_the_output_type(out_dtype):
+    """The partial slabs are sized by the PARTIAL dtype, not the O dtype.
+
+    Sizing them from O would carve one byte per element for an FP8 output and
+    hand the kernel a workspace half as big as the half-precision partials it
+    writes -- a silent overrun that the numerics above would not localize."""
+    from cudnn.sdpa.fwd.api_dsl import SdpaFwdDslSm100
+
+    if torch.cuda.get_device_capability() not in ((10, 0), (10, 3), (10, 7)):
+        pytest.skip("per-tensor FP8 prefill requires cc10.0 / cc10.3 / cc10.7")
+    b, h, s_q, s_kv, d, dev = 1, 8, 512, 16384, 128, "cuda"
+    mk = lambda *sh: (torch.randn(*sh, device=dev) * 0.5).to(torch.float8_e4m3fn)
```

Ruff (0.16.3)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `mk` as a `def`

(E731)